sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -189,6 +189,7 @@ case class InsertIntoHadoopFsRelationCommand(
       sparkSession.catalog.refreshByPath(outputPath.toString)
 
       if (catalogTable.nonEmpty) {
+        sparkSession.catalog.refreshTable(catalogTable.get.identifier.quotedString)
```

Comment:
btw, I thought the cost of refreshing the table can be quite high?

Reply:
It may be high now. it get `sizeInBytes` from:
https://github.com/apache/spark/blob/25c2776dd9ae3f9792048c78be2cbd958fd99841/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala#L88-L91

Reply:
shall we do
```
if (catalogTable.isDefined) {
  sparkSession.catalog.refreshTable(catalogTable.get.identifier.quotedString)
} else {
  // refresh cached files in FileIndex
       fileIndex.foreach(_.refresh())
       // refresh data cache if table is cached
      sparkSession.catalog.refreshByPath(outputPath.toString)
}
```

Reply:
shall we follow `InsertIntoHiveTable` and use `sparkSession.sessionState.catalog.refreshTable`?

Reply:
OK, Fixed.